packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations/RelationModal.tsx

```diff
@@ -337,7 +375,17 @@ const RelationModal = ({ children }: { children: React.ReactNode }) => {
                   type: 'GO_FULL_PAGE',
                 });
                 if (!state.hasUnsavedChanges) {
-                  navigate(getFullPageUrl(currentDocumentMeta));
+                  if (isCreating) {
+                    navigate(
+                      `/content-manager/${currentDocumentMeta.collectionType}/${currentDocumentMeta.model}/create${
+                        currentDocumentMeta.params?.locale
+                          ? `?plugins[i18n][locale]=${currentDocumentMeta.params.locale}`
+                          : ''
+                      }`
```

Comment:
It looks like you are computing this same value on link 484 could it be a function?